src/NHibernate/Hql/Ast/ANTLR/SqlGenerator.g

```diff
@@ -188,7 +188,7 @@ fromTable
 	// Write the table node (from fragment) and all the join fragments associated with it.
 	: ^( a=FROM_FRAGMENT  { Out(a); } (tableJoin [ a ])* )
 	| ^( a=JOIN_FRAGMENT  { Out(a); } (tableJoin [ a ])* )
-	| ^( a=ENTITY_JOIN    { Out(a); } )
+	| ^( a=ENTITY_JOIN    { Out(a); } (tableJoin [ a ])* )
```

Comment:
That fixes processing of fetches and joins on entity join (hibernate [commit](https://github.com/hibernate/hibernate-orm/commit/a2781e6654a940ba581a825bb165da0eb05092b2))